src/niftyone/figures/generator.py

```diff
@@ -104,22 +136,44 @@ def _figure_name(self) -> None:
             self.entities["extra_entities"]["figure"] = self.view_kwargs["figure"]
             del self.view_kwargs["figure"]
 
-    def generate(self, record: pd.Series, out_dir: Path, overwrite: bool) -> None:
+    def generate(
+        self,
+        records: list[pd.Series],
+        out_dir: Path,
+        overwrite: bool,
+    ) -> None:
         """Main call for generating view."""
         if not self.view_fn:
             raise ValueError("View is not provided, unable to create generator.")
 
         # Update figure name if necessary
         self._figure_name()
 
-        img_path = Path(record["finfo"]["file_path"])
+        img_path = Path(records[0]["finfo"]["file_path"])
         logging.info("Processing: %s", img_path)
+
         img = nib.nifti1.load(img_path)
         img = noimg.to_iso_ras(img)
 
-        existing_entities = BIDSEntities.from_dict(record["ent"])
+        # Handle overlays - currently only handles 1, update to handle multiple
+        overlays = []
+        # Temporary logic to handle multiple overlays
+        if len(records) > 2:
+            raise NotImplementedError("Multi-image overlay not yet implemented")
+
+        if len(records) > 1:
```

Comment:
Yeah, its a little confusing - hopefully we can get rid of this all together soon and have multi-image overlay supported. Its `>2` given that 2 records would suggest 1 main image and 1 overlay and anything above that = multiple overlays.

Reply:
Tried to rearrange it such that it is a little less confusing. 

1. check if `records > 1` - if so go into the conditional
2. (temporary) check if `records >2` inside conditional - if so throw an error since this means multiple overlays
3. loop inside conditional over remaining overlays

Given that we don't handle more than 2 records right now, the loop isn't strictly necessary, but I put this in here to deal with multiple overlays in the (near) future.